### PR TITLE
Fix nullable DB fields to set to null via the API

### PR DESCRIPTION
**Pull request by Manisha15**
Fix nullable DB fields which could not be set to null via the API

fixes #6908
https://pulp.plan.io/issues/6908

### CODE REVIEW

File: CHANGES/6908.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed nullable DB fields which could not be set to null via the API
```
Comment (mdellweg) [inline CHANGES/6908.bugfix:1]: I only see a single one, but maybe `architectures` should be treated that way, too.

Comment (Manisha15) [reply CHANGES/6908.bugfix:1]: I thought the ticket was for components only but if it was for all attributes I will fix them all.

Comment (quba42) [reply CHANGES/6908.bugfix:1]: Do we also need it for the `gpgkey` field? Is it a valid use case to add a gpg key to a remote and then later remove it again completely?

Comment (quba42) [inline CHANGES/6908.bugfix:1]: For the changelog entry, perhaps the following might be a bit clearer for users:
```suggestion
Fixed a bug where some nullable fields for remotes could not be set to null via the API.
```

Comment (mdellweg) [reply CHANGES/6908.bugfix:1]: Probably yes. "Des Menschen Wille ist sein Himmelreich." ("Someones will is his kingdom of heavens.")

**Comment by quba42** [comment]
Same here, we should still fixup/squish these two commits into one.

**Pull request merged**